File: Beecrowd/Beginner/1435 - 1985/1983 - The Chosen/C++/1983_TheChosen.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int students;
    cin >> students;

    int student[students];
    double grade[students];

    for (int i = 0; i < students; i++) {
        cin >> student[i] >> grade[i];
    }

    double highestGrade = grade[0];
    int highestGradeStudent = student[0];

    for (int i = 1; i < students; i++) {
        if (grade[i] > highestGrade) {
            highestGrade = grade[i];
            highestGradeStudent = student[i];
        }
    }

    if (highestGrade >= 8) {
        cout << highestGradeStudent << endl;
    } else {
        cout << "Minimum note not reached" << endl;
    }

    return 0;
}

File: Beecrowd/Beginner/1435 - 1985/1983 - The Chosen/Python/1983_TheChosen.py
students = int(input())

student = []
grade = []

for i in range(students):
    s, g = map(float, input().split())
    student.append(int(s))
    grade.append(g)

highest_grade = grade[0]
highest_grade_student = student[0]

for i in range(1, students):
    if grade[i] > highest_grade:
        highest_grade = grade[i]
        highest_grade_student = student[i]

if highest_grade >= 8:
    print(highest_grade_student)
else:
    print("Minimum note not reached")
